packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -868,7 +1001,8 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
                   } catch (...) {
                     TMPL::asyncMethodCallExecutionFail(
                         moduleName, methodName, id);
-                    throw;
+                      auto exception = std::current_exception();
+                    rejectWithException(rejectJSIFunctionWeakWrapper, exception, jsInvocationStack);
```

Comment:
This needs gating, as it fundamentally changes how we handle exceptions thrown in async Java code.

Reply:
`rejectTurboModulePromiseOnNativeError` added.